lib/puppet/facter_impl.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+#
+# @api private
+# Default Facter implementation that delegates to Facter API
+#
+
+module Puppet
+  class FacterImpl
+    def initialize
+      require 'facter'
+    end
+
+    def value(fact_name)
+      ::Facter.value(fact_name)
+    end
+
+    def add(name, &block)
+      ::Facter.add(name, &block)
+    end
+  end
```

Comment:
Do we need to implement the other methods?

```
$ git grep -o "Facter\.\w*" lib | cut -d':' -f2 | sort | uniq
Facter.
Facter.add
Facter.clear
Facter.debugging
Facter.load_external
Facter.on_message
Facter.reset
Facter.resolve
Facter.respond_to
Facter.search
Facter.search_external
Facter.to_hash
Facter.trace
Facter.value
```

Reply:
Ah I see that's handled in the other PR: https://github.com/puppetlabs/puppet/pull/8749/commits/04b4318725bab3494cf634ce9520b1b2690a7c00